src/App.tsx

```diff
@@ -14,11 +15,97 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[],
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  add = (item: string) => {
+    this.setState((prev) => {
+      prev.selectedGoods.push(item);
+
+      return { selectedGoods: prev.selectedGoods };
+    });
+  };
+
+  remove = (item: string) => {
+    const indexOf = this.state.selectedGoods.indexOf(item);
+
+    this.setState((prev) => {
+      prev.selectedGoods.splice(indexOf, 1);
+
+      return { selectedGoods: prev.selectedGoods };
+    });
+  };
+
+  title = () => {
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case (0):
+        return 'No goods selected';
+
+      case (1):
+        return `${selectedGoods[0]} is selected`;
+
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')}
+          and ${selectedGoods.slice(-1)} are selected`;
+    }
+  };
+
+  render() {
+    return (
+      <div className="app">
+        <div>
+          <h1 className="app__title">{this.title()}</h1>
+
+          <button
+            className="app__clear"
+            type="button"
+            onClick={() => {
+              this.setState({ selectedGoods: [] });
+            }}
+          >
+            {this.state.selectedGoods.length > 0 && 'Clear'}
+          </button>
+        </div>
+
+        <ul className="app__list">
+          {goodsFromServer.map(item => (
+            <li
```

Comment:
you can check includes and after use isGoodsIncludeItem value 

```suggestion
          {goodsFromServer.map(item => {
            const isGoodsIncludeItem = this.state.selectedGoods.includes(item);
    
            return (
            <li
```